packages/account-tree-controller/src/AccountTreeController.ts

```diff
@@ -316,6 +323,37 @@ export class AccountTreeController extends BaseController<
     }
   }
 
+  #handleAccountRenamed(account: InternalAccount) {
+    if (!isEvmAccountType(account.type)) {
+      return;
+    }
+
+    const context = this.#accountIdToContext.get(account.id);
+
+    if (context) {
+      const { walletId, groupId } = context;
+
+      const wallet = this.state.accountTree.wallets[walletId];
+      if (wallet) {
+        const group = wallet.groups[groupId];
+        if (group) {
+          const isDefaultNameRegex = /^Account ([0-9]+)$/u;
```

Comment:
Can you put this in the top-level like `DEFAULT_ACCOUNT_NAME_REGEX` maybe?

I'll aso add a quick comment explaining that we both use the same naming convention for BIP-44 accounts and legacy HD accounts.

Reply:
Also, I think we can add a comment explaining we only consider HD accounts and private key accounts for now.

Because `isEvmAccountType` would also be true for private keys/hardware wallet accounts for example.

But this regex will not match those. (Unless we're re-using the custom prefix we have with `keyringToTypeNames` which is used by the `AccountsController` default naming logic)